net/tcp/tcp_sendfile.c

```diff
@@ -274,6 +276,8 @@ static uint16_t sendfile_eventhandler(FAR struct net_driver_s *dev,
 
   else if ((flags & TCP_REXMIT) != 0)
     {
+      uint32_t sndlen;
```

Comment:
If we strictly follow C89, we can't even use uint32_t since it define in C99.

Reply:
NuttX defines stdint related types by itself:

nuttx/include/stdint.h:
`typedef _uint32_t           uint32_t;
`
nuttx/arch/z80/include/z80/types.h:
`typedef unsigned long      _uint32_t;`

Reply:
Yes, that's why I suggest that we only need to conform c89 core language spec, not the whole c89 spec. We have to stick to c89 just because some arch compiler can't support the c99 grammar. From the language perspective, c99 is much better than c89, so we should leverage the new 99 feature which not relate to the grammar(such as the platform independent type).

Reply:
> Yes, that's why I suggest that we only need to conform c89 core language spec, not the whole c89 spec. We have to stick to c89 just because some arch compiler can't support the c99 grammar. From the language perspective, c99 is much better than c89, so we should leverage the new 99 feature which not relate to the grammar(such as the platform independent type) as much as possible.

I agree. So what is the issue with the highlighted line (uint32_t sndlen)? What is your suggestion?

Reply:
I suggest that the real change we need is move the variable definition in the middle to the begin, other change could be safely ignore.

Reply:
I think here just was a highlight to say that `%z` follows the similar case as stdint types.

Reply:
Do you mean we should do not care about these two types of warnings?
warning: format ‘%p’ expects argument of type ‘void *’, but argument 3 has type ‘struct tcp_conn_s *’ [-Wformat=]
warning: ISO C90 does not support the ‘z’ gnu_printf length modifier [-Wformat=]

However, we should care only about the following warning type:
warning: ISO C90 forbids mixed declarations and code [-Wdeclaration-after-statement]

Right?

Reply:
Yes, you are right. The last issue will become error in the old compiler. But, the first two aren't.

Reply:
Good. I have updated the patch.